gemini/sample-apps/genwealth/function-scripts/analyze-prospectus/main.py

```diff
@@ -0,0 +1,151 @@
+"""Cloud Function code to analyze a prospectus"""
+
+import base64
+import os
+
+import functions_framework
+from google.cloud.alloydb.connector import Connector
+from langchain_core.prompts import PromptTemplate
+from langchain_google_vertexai import VertexAI
+import sqlalchemy
+
+
+# Triggered from a message on a Cloud Pub/Sub topic.
+@functions_framework.cloud_event
+def analyze_prospectus(cloud_event):
+    """Function to analyze prospectus"""
+    # Print out the data from Pub/Sub, to prove that it worked
+    ticker = base64.b64decode(cloud_event.data["message"]["data"])
+    ticker = ticker.decode("utf-8")
+    print(ticker)
+
+    # Environment Vars
+    region = os.environ["REGION"]
+    project_id = os.environ["PROJECT_ID"]
+
+    # AlloyDB Vars
+    cluster = "alloydb-cluster"
+    instance = "alloydb-instance"
+    database = "ragdemos"
+    table_name = "langchain_vector_store"
+    user = "postgres"
+    password = os.environ["ALLOYDB_PASSWORD"]
+
+    # Setup sync connector
+    connector = Connector()
+
+    def getconn():
+        conn = connector.connect(
+            f"projects/{project_id}/locations/{region}/clusters/{cluster}/instances/{instance}",
+            "pg8000",
+            user=user,
+            password=password,
+            db=database,
+        )
+        return conn
+
+    # create connection pool
+    pool = sqlalchemy.create_engine(
+        "postgresql+pg8000://",
+        creator=getconn,
+    )
+
+    # Prep SQL statement
+    sql = f"SELECT content FROM {table_name} WHERE ticker = '{ticker}' ORDER BY page, page_chunk"
+
+    # Prep model and template
+    model = VertexAI(
+        model_name="text-bison@002", max_output_tokens=1024, temperature=0.0
```

Comment:
Could this use Gemini instead of text-bison?

Reply:
Yes, updated.